realm/realm-library/src/main/cpp/io_realm_internal_TableQuery.cpp

```diff
@@ -84,12 +84,17 @@ static TableRef getTableForLinkQuery(jlong nativeQueryPtr, JniLongArray& tablesA
 }
 
 // Return TableRef point to original table or the link table
-static TableRef getTableByArray(jlong nativeQueryPtr, JniLongArray& indicesArray)
+static TableRef getTableByArray(jlong nativeQueryPtr, JniLongArray& tablesArray, JniLongArray& indicesArray)
 {
     TableRef table_ref = Q(nativeQueryPtr)->get_table();
     jsize link_element_count = indicesArray.len() - 1;
     for (int i = 0; i < link_element_count; i++) {
-        table_ref = table_ref->get_link_target(size_t(indicesArray[i]));
+        if (TBL(tablesArray[i]) == nullptr) {
```

Comment:
So, I can say with absolute certainty, that there is still a boatload of use of those macros.  It is doubly confusing to have both techniques just a couple lines apart (... and I don't recall if that is the case, here).  If we are going to "move", this is one of the few cases that I would suggest we not do it piecemeal.  We should change the style, at the very least, for an entire file.